scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx

```diff
@@ -17,46 +21,71 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
     dropdownElement: any;
 
     componentDidMount() {
-        this.stickyElementTracker = new StickElementsWithTracking(this.props.targetElement, this.dropdownElement);
+        this.stickyElementTracker = new StickElementsWithTracking(
+            this.props.targetElement,
+            this.dropdownElement,
+        );
     }
     componentWillUnmount() {
         this.stickyElementTracker.destroy();
     }
 
+    onSuggestionClick(suggestion: ISpellcheckerSuggestion) {
+        this.props.dispatch(
+            actions.replaceWord({
+                word: {
+                    text: this.props.warning.text,
+                    offset: this.props.warning.startOffset,
+                },
+                newWord: suggestion.text,
+            }),
+        );
+    }
+
     render() {
-        const {suggestions} = this.props.warning;
-        const {spellchecker} = this.props;
+        const { suggestions, message } = this.props.warning;
+        const { spellchecker } = this.props;
+
+        // If the message exists, use it as the button text
+        // instead of the suggestion
+        const messageExists = Boolean(message);
 
         return (
             <div className={'dropdown open suggestions-dropdown'}
                 ref={(el) => this.dropdownElement = el}
                 style={{zIndex: 999, border: 'solid transparent', borderWidth: '6px 0'}}
                 data-test-id="spellchecker-menu"
             >
-                <ul className={'dropdown__menu'} style={{position: 'static'}}>
-                    <div className="form-label" style={{margin: '0 16px'}}>{gettext('Suggestions')}</div>
-                    {
-                        suggestions.length === 0
-                            ? <li><button>{gettext('SORRY, NO SUGGESTIONS.')}</button></li>
-                            : suggestions.map((suggestion, index) =>
+                <ul className={'dropdown__menu'} style={{ position: 'static' }}>
+                    <div className='form-label' style={{ margin: '0 16px' }}>
+                        {gettext('Suggestions')}
+                    </div>
+                    {messageExists && (
```

Comment:
I see you've learned some angular :smile: 

You should never write multiple independent `if` statements for what is actually an `if/else`. When `if/else` is used, it's immediately clear that only one condition can run. When you use independent `if` statements, you remove that link and developer needs to analyze the conditions to understand that you really meant to write an `if/else`.

Use ternaries instead:

```
{
    messageExists
        ? (
            <div>a</div>
        )
        : (
            suggestions.length === 0
                ? (<div>a</div>)
                : (<div>b</div>)
        )
}
```

Nested ternaries might not look the nicest at the first glance either, but at least they represent the logic correctly.

Reply:
At first it was not a ternary operator cause I didn't have the `!messageExist` part (it was showing the suggestions even if the message existed) so this was necessary. But yeah when I added the `!messageExists` part I didn't see the big picture